source/simulator/core.cc

```diff
@@ -2569,6 +2569,10 @@ namespace aspect
       }
     else
       {
+        time                      = parameters.start_time;
+        timestep_number           = 0;
+        time_step = old_time_step = 0;
+
```

Comment:
Is this move important or just by accident?

Reply:
This is important, because time needs to be initialized before the refinement happens.

Reply:
But none of the initial topography models should be time dependent. If they are: Is this currently a bug, or has something changed in this PR that makes this necessary?

Reply:
If I remember correctly, it has to do with the AsciiDataBoundary. The refinement function calls this because it needs to know the position and it ends up in utilities:1859
```
    template <int dim>
    double
    AsciiDataBoundary<dim>::
    get_data_component (const types::boundary_id             boundary_indicator,
                        const Point<dim>                    &position,
                        const unsigned int                   component) const
    {
      if (this->get_time() - first_data_file_model_time >= 0.0)
```
If the time is not set, the default value of numbers::signaling_nan<double>() will be triggered. This is because the ascii data was originally made for time dependent boundary conditions (or that is what I can make from reading the code) and therefore needs the current time to decide what the current boundary conditions are. Now that that code is also used for initial topography, it is getting called earlier then it was designed for.

Reply:
I see. Then this makes sense. Thanks for the explanation.